patroni/ctl.py

```diff
@@ -828,6 +828,80 @@ def members(obj, cluster_names, fmt, watch, w, extended, ts):
             cluster = dcs.get_cluster()
             output_members(cluster, cluster_name, extended, fmt)
 
+def output_topology(cluster, name):
+    res = []
+    masters=[]
+    topology = {}
+    filler = ' '
+    initialize = {None: 'uninitialized', '': 'initializing'}.get(cluster.initialize, cluster.initialize)
+    cluster = cluster_as_json(cluster)
+    members = {member['host']:member for member in [m for m in cluster['members'] if 'host' in m]}
+    for _, data in members.items():
+        if data['master_host'] is None:
+            masters.append(data)
+        else:
+            if data['master_host'] not in topology:
+                topology[data['master_host']] = []
```

Comment:
`topology.setdefault(data['master_host'], [])` construction may be used as simplified version of 

`
if data['master_host'] not in topology:
     topology[data['master_host']] = []
`

Reply:
fixed